From 4d0343a386d6e3de4a15ae33b591d947359eeec2 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 18 Aug 2022 19:59:33 +0300
Subject: [PATCH] [SW-86869] gaudi3: put arc in sleep state

Currently, rom executes while loop once it reaches at end of
recovery. This patch add change to put ARC in sleep state.

Change-Id: I3d18e28e81f6bb87645f3f56af0ad07ca9ad176b
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 60cd80e..007cddb 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -254,8 +254,8 @@
 	/*
 	 * ROM Execution is expected to reach here only in case EROM image
 	 * download and verification error.
-	 * Stay here and do not return
+	 * Stay here (sleep) and do not return
 	 */
 	while (1)
-		;
+		wfe();
 }
